service/raft/raft_group_registry.cc

```diff
@@ -117,7 +118,24 @@ class gossiper_state_change_subscriber_proxy: public gms::i_endpoint_state_chang
 
     virtual future<>
     on_alive(gms::inet_address endpoint, gms::endpoint_state ep_state) override {
-        return on_endpoint_change(endpoint, ep_state);
+        co_await utils::get_local_injector().inject_with_handler("raft_group_registry::on_alive", [endpoint, &ep_state] (auto& handler) -> future<> {
+            auto app_state_ptr = ep_state.get_application_state_ptr(gms::application_state::HOST_ID);
+            if (!app_state_ptr) {
+                co_return;
+            }
+            
+            raft::server_id id(utils::UUID(app_state_ptr->value()));
+            rslog.info("gossiper_state_change_subscriber_proxy::on_alive() {} {}", endpoint, id);
+            auto second_node_ip = handler.get("second_node_ip");
+            assert(second_node_ip);
+
+            if (endpoint == gms::inet_address(sstring{*second_node_ip})) {
+                rslog.info("Sleeping before handling on_alive");
+                co_await handler.wait_for_message(std::chrono::steady_clock::now() + std::chrono::seconds{6});
```

Comment:
Please increase this timeout to 1 minute, we don't want the test to be flaky on slow machines.